src/utils/state.ts

```diff
@@ -73,9 +74,17 @@ export const useStateWithDeps = <Data, Error, S = State<Data, Error>>(
     []
   )
 
-  // Always update the state reference.
   useIsomorphicLayoutEffect(() => {
-    stateRef.current = state
+    // Always update the state reference if it's not used as the initial state
+    // only.
+    if (!asInitialState) {
```

Comment:
Do you feel it make sense to move this auto update of `stateRef.current` outside of `useStateWithDeps` ?

We do auto update in use-swr.ts but not in mutation.ts

Reply:
Yeah makes sense!